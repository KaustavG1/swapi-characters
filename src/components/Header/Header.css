.header {
  display: flex;
}

.fav-button {
  font-size: 1.9rem;
}
