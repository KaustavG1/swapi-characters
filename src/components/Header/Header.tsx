import { useNavigate } from "react-router";
import Button from "../common/Button/Button";
import SearchBox, { SearchBoxProps } from "../SearchBox/SearchBox";
import "./Header.css";

function Header({searchTerm, setSearchTerm}: SearchBoxProps) {
  const navigate = useNavigate();

  const handleFavClick = () => {
    navigate("/favourites");
  };

  return (
    <div className="header">
      <SearchBox searchTerm={searchTerm} setSearchTerm={setSearchTerm} />
      <Button icon="♥️" text="" className="fav-button" onClick={handleFavClick} />
    </div>
  );
}

export default Header;
