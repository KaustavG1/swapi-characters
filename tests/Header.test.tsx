import { it, expect, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import Header from "../src/components/Header/Header";
import { BrowserRouter } from "react-router";

describe("Header component", () => {
  it("renders 'Header' component", () => {
    const setSearchTerm = vi.fn();
    const { container } = render(
      <BrowserRouter>
        <Header searchTerm="test search term" setSearchTerm={setSearchTerm} />
      </BrowserRouter>
    );
    const searchBar = screen.getByPlaceholderText(/search characters/i);
    const button = screen.getByText(/♥️/);
    expect(searchBar).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    fireEvent.change(searchBar, { target: { value: "new value" } });
    expect(setSearchTerm).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });
});
