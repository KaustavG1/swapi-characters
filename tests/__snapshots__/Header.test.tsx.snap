// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders 'Header' component 1`] = `
<div>
  <div
    class="header"
  >
    <input
      class="search-bar"
      placeholder="search characters"
      type="text"
      value="test search term"
    />
    <div>
      <button
        class="fav-button button-default"
      >
        <span>
          ♥️
        </span>
        <span />
      </button>
    </div>
  </div>
</div>
`;
